src_common/mem_utils.c

```diff
@@ -0,0 +1,60 @@
+#ifdef HAVE_DYN_MEM_ALLOC
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include "mem.h"
+#include "mem_utils.h"
+
+/**
+ * Format an unsigned number up to 32-bit into memory into an ASCII string.
+ *
+ * @param[in] value Value to write in memory
+ * @param[out] length number of characters written to memory
+ *
+ * @return pointer to memory area or \ref NULL if the allocation failed
+ */
+char *mem_alloc_and_format_uint(uint32_t value, uint8_t *const length) {
+    char *mem_ptr;
+    uint32_t value_copy;
+    uint8_t size;
+
+    size = 1;  // minimum size, even if 0
+    value_copy = value;
+    while (value_copy >= 10) {
+        value_copy /= 10;
+        size += 1;
+    }
+    // +1 for the null character
+    if ((mem_ptr = mem_alloc(sizeof(char) * (size + 1)))) {
+        snprintf(mem_ptr, (size + 1), "%u", value);
+        mem_dealloc(sizeof(char));  // to skip the null character
```

Comment:
I would agree but since the length is explicitly given to the function, there is no way for the caller to not know how long the formatted string will be.